modules/miner.go

```diff
@@ -14,6 +15,15 @@ type MinerInfo struct {
 	Address        types.UnlockHash
 }
 
+// MinerWork is all the information a miner needs to do work
+type MinerWork struct {
+	Block      types.Block
+	ParentID   types.BlockID
+	Nonce      uint64
+	MerkleRoot crypto.Hash
+	Target     types.Target
+}
```

Comment:
There is a catch, the nonce is only 64 bits. This isn't really a problem while FPGA mining but when ASICs start using the api they will be close to using all 64 bits.

The fix to that is to have 'getwork' always return a unique Merkle root, which it achieves by putting a single arbitrary data transaction in the block and manipulating the string for the arbitrary data. Relatively easy to implement.

Then the asics just call 'getwork' again after using up all 64 bits and are guaranteed to have no overlap on their blocks.